HPS/Heartland/view/adminhtml/templates/form/cc.phtml

```diff
@@ -1,151 +1,285 @@
  */
 
 // @codingStandardsIgnoreFile
 /**
- * @var \Magento\Payment\Block\Adminhtml\Transparent\Form $block
+ * @var \HPS\Heartland\Block\Form\Cc $block
  */
+
+$hpsTokens = $block->getCcTokens();
+$pubApiKey = \HPS\Heartland\Helper\Data::getPublicKey();
 $code = $block->escapeHtml($block->getMethodCode());
 $ccType = $block->getInfoData('cc_type');
 $ccExpMonth = $block->getInfoData('cc_exp_month');
-$ccExpYear = $block->getInfoData('cc_exp_year');
+$ccExpYear = $block->getInfoData('cc_exp_year'); //
+//var_dump($hpsTokens);
 ?>
-<fieldset class="admin__fieldset payment-method" id="payment_form_<?php /* @noEscape */ echo $code; ?>"
+<fieldset class="admin__fieldset payment-method" id="payment_form_<?php /* @noEscape */
+echo $code; ?>"
           style="display:none">
+
+    <script>
+        var <?php /* @noEscape */ echo $code; ?>;
+        require([
+                'jquery'
+            ],
+
+            /*
+             *  Heartland payment method model
+             *
+             *  @category    HPS
+             *  @package     HPS_Heartland
+             *  @author      Charlie Simmons <charles.simmons@e-hps.com>
+             *  @copyright   Heartland (http://heartland.us)
+             *  @license     https://github.com/hps/heartland-magento2-extension/blob/master/LICENSE.md
+             */
+
+            /*browser:true*/
+            /*global define*/
+            function ($) {
+                'use strict';
+                /**
+                 * Get the public key from HPS/Heartland/Controller/Hss/Pubkey.php as url configured based on HPS/Heartland/etc/frontend/routes.xml
+                 * if there is any form of error we disable the payment form
+                 *
+                 */
+                $(document).ready(function ($) {
+                    $("#iframes").fadeOut();
+                    <?php /* @noEscape */ echo $code; ?> = {
+
+                        _HPS_setHssTransaction1: function (response) {//{"token_value":"3","cc_last4":"1111","cc_type":"visa","cc_exp_month":"2","cc_exp_year":"2025"}
+                            if(document.querySelector("#hssCardSelected" + response.token_value.trim()))
+                            {
+                                $("#iframes").fadeOut();
+                                document.querySelector("#hssCardSelected" + response.token_value.trim()).checked = true;
+                                document.querySelector('#securesubmit_token').value = response.token_value.trim();
+
+                                document.querySelector('#hps_heartland_cc_number').value = response.cc_last4.trim();
+                                document.querySelector('#hps_heartland_cc_type').value = response.cc_type.trim();
+                                document.querySelector('#hps_heartland_expiration').value = response.cc_exp_month.trim();
+                                document.querySelector('#hps_heartland_expiration_yr').value = response.cc_exp_year.trim();
```

Comment:
query selector replaced by jquery, so it's not required to test the return value